spec/unit/provider/package/gem_spec.rb

```diff
@@ -185,6 +185,71 @@
         end
       end
     end
+
+    describe 'insync?' do
+      describe 'for array of versions' do
+        let(:is) { ['1.3.4', '3.6.1', '5.1.2'] }
+
+        it 'returns true for ~> 1.3' do
+          resource[:ensure] = '~> 1.3'
+          expect(provider.insync?(is)).to eq(true)
```

Comment:
If the expectation fails we get a message like:

```
       expected: false
            got: true
```

so we usually recommend using the predicate matchers, e.g. `expect(provider).to be_insync(is)` as it supposed to provide a better error message when it fails, something along the lines of:

```
expected ....insync?(["1.3.4", "3.6.1", "5.1.2"])` to return false, got true
```

But it looks like rspec3 is including `provider.inspect` in the error message instead of `provider.to_s`, making overly verbose error messages:

```
     Failure/Error: expect(provider).to be_insync(is)
       expected `#<Puppet::Type::Package::ProviderGem:0x007fe968ce5af0 @property_hash={}, @resource=#<Puppet::Type::Package:0x007fe968ce7a80 @parameters={:provider=>#<Puppet::Type::Package::ParameterProvider:0x007fe968ce6d10 @resource=#<Puppet::Type::Package:0x007fe968ce7a80 ...>, @value=:pkgdmg>, 
:name=>#<Puppet::Type::Package::ParameterName:0x007fe968ce68b0 @resource=#<Puppet::Type::Package:0x007fe968ce7a80 ...>, @value="myresource">, 
:ensure=>#<Puppet::Type::Package::Ensure:0x007fe968ce67c0 @resource=#<Puppet::Type::Package:0x007fe968ce7a80 ...>, @shouldorig=["~> 1.3"], @should=["~> 1.3"]>, 
:configfiles=>#<Puppet::Type::Package::ParameterConfigfiles:0x007fe968ce64f0 @resource=#<Puppet::Type::Package:0x007fe968ce7a80 ...>, @value=:keep>, 
:reinstall_on_refresh=>#<Puppet::Type::Package::ParameterReinstall_on_refresh:0x007fe968ce5f50 @resource=#<Puppet::Type::Package:0x007fe968ce7a80 ...>, @value=:false>, 
:loglevel=>#<Puppet::Type::MetaParamLoglevel:0x007fe968ce5d20 @resource=#<Puppet::Type::Package:0x007fe968ce7a80 ...>, @value=:notice>}, 
@title="myresource", @tags=#<Puppet::Util::TagSet: {"package", "myresource"}>, 
@original_parameters={:name=>"myresource", :ensure=>:installed}, @name_var_cache=false, 
@provider=#<Puppet::Type::Package::ProviderGem:0x007fe968ce5af0 ...>>>.insync?(["1.3.4", "3.6.1", "5.1.2"])` to return false, got true
```

It appears this rspec-expectations change "Improve the failure message for be_xyz predicate matchers so that it includes the inspect output of the receiver. (Erik Michaels-Ober, Sam Phippen)" is the cause. Thoughts @peterhuene?